homeassistant/components/linkplay/utils.py

```diff
@@ -44,3 +54,15 @@ def get_info_from_project(project: str) -> tuple[str, str]:
             return MANUFACTURER_IEAST, MODELS_IEAST_AUDIOCAST_M5
         case _:
             return MANUFACTURER_GENERIC, MODELS_GENERIC
+
+
+async def async_get_client_session(hass: HomeAssistant) -> ClientSession:
+    """Get a ClientSession that can be used with LinkPlay devices."""
+    hass.data.setdefault(DOMAIN, {})
+    if CONF_SESSION not in hass.data[DOMAIN]:
+        clientsession: ClientSession = await async_create_unverified_client_session()
```

Comment:
Yea that was the thing I was talking about in the library bump. That means that the code is doing blocking IO to open the file on the file system and get the cert

Reply:
As there is no native async function for `load_cert_chain` I assumed HA would allow it. I'll edit the code in the library to wrap it in a `run_in_executor`